snowflake-api/src/lib.rs

```diff
@@ -225,22 +245,53 @@ impl SnowflakeApi {
             .with_token(info.creds.aws_token)
             .build()?;
 
+        let s3_mutex = Arc::new(Mutex::new(s3));
+
         // todo: security vulnerability, external system tells you which local files to upload
+        let mut final_source_locs: Vec<String> = Vec::new();
         for src_path in src_locations {
-            let path = Path::new(src_path);
-            let filename = path
-                .file_name()
-                .ok_or(SnowflakeApiError::InvalidLocalPath(src_path.clone()))?;
-
-            // fixme: unwrap
-            let dest_path = format!("{}{}", bucket_path, filename.to_str().unwrap());
-            let dest_path = object_store::path::Path::parse(dest_path)?;
-
-            let src_path = object_store::path::Path::parse(src_path)?;
+            for entry in glob::glob(src_path).expect("Failed to read glob pattern") {
+                match entry {
+                    Ok(path) => match path.to_str() {
+                        Some(item) => final_source_locs.push(item.to_string()),
+                        None => return Err(SnowflakeApiError::InvalidLocalPath(src_path.clone())),
+                    },
+                    Err(_) => return Err(SnowflakeApiError::InvalidLocalPath(src_path.clone())),
+                }
+            }
+        }
 
-            let fs = LocalFileSystem::new().get(&src_path).await?;
+        let bucket_path = bucket_path.to_string();
+        let mut set = JoinSet::new();
+        for src_path in final_source_locs {
```

Comment:
I think using Go and .NET official implementations as reference is a good idea here. 
https://github.com/snowflakedb/snowflake-connector-net/blob/7d478f0eaeb71493b878a47f87193df4e69c39b2/Snowflake.Data/Core/FileTransfer/SFFileTransferAgent.cs#L205-L217

The application should take into account if parallel uploading is supported for given target and obey threshold:
https://github.com/mycelial/snowflake-rs/blob/main/snowflake-api/src/responses.rs#L197-L201

Reply:
So for right now I am using the Snowflake recommended limits of 64MB for file size and 4 threads for concurrency. I did not see a way to easily get the values you have linked here into the function, do you have any suggestions?